function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs

```diff
@@ -204,15 +242,28 @@ private async Task<bool> JoinEventData(Event olEvent, string jobRunId)
                 }
                 olEvent.Run.Facets.EnvironmentProperties = envFacet;
 
-                // clean up table over time
-                try
-                {
-                    var delresp = await _tableClient.DeleteEntityAsync(TABLE_PARTITION, olEvent.Run.RunId);
-                }
-                catch (Exception ex)
-                {
-                    _log.LogError(ex, $"OlMessageConsolodation-JoinEventData: Error {ex.Message} when deleting entity");
-                }
+            // Add Inputs to event if not already there (will only be done for DataSourceV2 sources)
+            if (olEvent.Inputs.Count == 0) {
```

Comment:
Is there a case where there may be an input present initially and then more inputs later on?

What about a scenario where ABFSS is used and joined with a Data Source V2 source like Cosmos?